Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp

```diff
@@ -471,20 +463,16 @@ void IntegratePeaksMD2::integrate(typename MDEventWorkspace<MDE, nd>::sptr ws) {
         bgSignal *= scaleFactor;
         bgErrorSquared *= scaleFactor * scaleFactor;
       } else {
-        for (size_t j = 0; j < numSteps; j++) {
-          wsProfile2D->dataX(i)[j] = static_cast<double>(j);
-          wsProfile2D->dataY(i)[j] = signal_fit[j];
-          wsProfile2D->dataE(i)[j] = std::sqrt(signal_fit[j]);
-        }
+        setSpectrum(wsProfile2D, i, signal_fit);
```

Comment:
Covered in previous comments. Shouldn't need `setSpectrum` unless it returns a histogram.